types/time_test.go

```diff
@@ -63,13 +66,14 @@ func TestUnmarshalJSON(t *testing.T) {
 	require.ErrorIs(t, json.Unmarshal([]byte(`"1606292218213.45.8"`), &testTime), strconv.ErrSyntax)
 }
 
-// 5030734	       240.1 ns/op	     168 B/op	       2 allocs/op (current)
-// 2716176	       441.9 ns/op	     352 B/op	       6 allocs/op (previous)
+// 5697680	       203.8 ns/op	     168 B/op	       2 allocs/op (current)
+// 5030734	       240.1 ns/op	     168 B/op	       2 allocs/op (previous)
 func BenchmarkUnmarshalJSON(b *testing.B) {
 	var testTime Time
 	for b.Loop() {
-		err := json.Unmarshal([]byte(`"1691122380942.173000"`), &testTime)
-		require.NoError(b, err)
+		if err := json.Unmarshal([]byte(`"1691122380942.173000"`), &testTime); err != nil {
+			require.NoError(b, err)
```

Comment:
```suggestion
			b.Fatal(err) 
```
Since in this case we've opted the orig style for perf, we can just default the whole thing to it to bypass require's overhead as we already know an err has occurred. This is probably a good policy only to use for benchmarks so we get a more accurate benchmark